python/lsst/daf/butler/registry/datasets/byDimensions/_storage.py

```diff
@@ -201,20 +243,38 @@ def certify(self, collection: CollectionRecord, datasets: Iterable[DatasetRef],
             raise TypeError(f"Cannot certify into collection '{collection}' "
                             f"of type {collection.type.name}; must be CALIBRATION.")
         tsRepr = self._db.getTimespanRepresentation()
-        protoRow = {
+        summaryRow = {
             self._collections.getCollectionForeignKeyName(): collection.key,
             "dataset_type_id": self._dataset_type_id,
         }
         rows = []
+        governorValues: Dict[str, Set[str]] = {
+            name: set() for name in self.datasetType.dimensions.governors.names
+        }
         dataIds: Optional[Set[DataCoordinate]] = set() if not tsRepr.hasExclusionConstraint() else None
         for dataset in datasets:
-            row = dict(protoRow, dataset_id=dataset.getCheckedId())
+            row = dict(summaryRow, dataset_id=dataset.getCheckedId())
             for dimension, value in dataset.dataId.items():
                 row[dimension.name] = value
             tsRepr.update(timespan, result=row)
+            for governorName, values in governorValues.items():
+                values.add(dataset.dataId[governorName])  # type: ignore
             rows.append(row)
             if dataIds is not None:
                 dataIds.add(dataset.dataId)
+        # Update the summary tables for this collection in case this is the
+        # first time this dataset type or these governor values will be
+        # inserted there.
+        self._db.ensure(self._summaries.datasetType, summaryRow)
```

Comment:
Is this the third time this identical logic has appeared in the one file? Can we not have a helper method in here?